src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -534,7 +534,7 @@ def get_op_handler(self, operation, operation_group=None):
         from azure.cli.core.profiles._shared import get_versioned_sdk_path
 
         for rt in AZURE_API_PROFILES[self.cli_ctx.cloud.profile]:
-            if operation.startswith(rt.import_prefix):
+            if operation.startswith(rt.import_prefix + '.'):
```

Comment:
While this does handles some scenarios (please provide some real use cases in the PR description), this still can't handle the keyvault track 1 and 2 SDKs' compatibility:

Track 1 SDK:
- `azure.keyvault`

Track 2 SDK:
- `azure.keyvault.secrets`
- `azure.keyvault.keys`

`azure.keyvault.secrets.SecretClient.xxx` and `azure.keyvault.keys.KeyClient.xxx` will be matched as Track 1 SDK `azure.keyvault.`.

Reply:
the change itself looks resolving the issue mentioned in description, same question on is there multiple namespaces containing dot after first dot? such as `azure.multipleapi.storage.blob`, `azure.multipleapi.storage.file`

Reply:
@jiasli If `azure.keyvault` is totally replaced with submodules such as `azure.keyvault.secrets` and `azure.keyvault.keys`, there will be no problem. 
But if the resource type for `azure.keyvault` still exists with new resource types with `azure.keyvault.secrets` and `azure.keyvault.keys`, it will have some problem.

@yungezz There are packages as you mentioned, e.g. `azure.mgmt.resource` in https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L44-L49 .

Reply:
https://github.com/Azure/azure-cli/blob/31a9724478c67751ae0bee6cc0d9b75b763df17c/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L44-L49

This doesn't have problem because there is no `azure.mgmt.resource` that can cause the same conflict as `azure.keyvault`. 

I am ok with this for now, but still we need to figure out a better solution.

Reply:
Yes. If you want to using submodules for `keyvault`, you cannot use `azure.keyvault` any more.